modules/concepts/hooks/list-of-hooks/moduleRoutes.md

```diff
@@ -29,4 +37,104 @@ This hook has a `$check_exception` parameter set to `false` (check permission ex
 
 ```php
 Hook::exec('moduleRoutes', ['id_shop' => $id_shop], null, true, false)
+```
+
+## Example implementation
+
+This hook must return an `array` of routes, keyed by the names of the routes, and each route structured by:
+
+```php
+array(
+  'rule' => string,
+  'keywords' => array(),
+  'controller' => string,
+  'params' => array(
+    'fc' => string,
+    'module' => string
+  )
+)
+```
+
+### `rule` and `keywords` parameters
+
+`rule` is the URL that will be matched by the router. `rule` can contain parameters, enclosed with `{}`. 
+For example, to add an `id` parameter in your route, use the following `rule`: 
+
+`myrule/{id}`
+
+Multiple parameters can be used, for example: 
+
+`myrule/{id}/{slug}`
+
+Each parameter in your `rule` must be declared in the `keywords` array.
+
+In our example, `id` must be an integer and `slug` a string, and will be forwarded to the controller with `id` and `slug` parameter names: 
+
+```php
+[
+  'rule' => 'myrule/{id}/{slug}',
+  'keywords' => [
+    'id' => [
+      'regexp' => '[0-9]*',
+      'param' => 'id'
+    ],
+    'slug' => [
+      'regexp' => '.*',
+      'param' => 'slug'
+    ]
+  ],
+  'controller' => 'myrulecontroller',
+  'params' => [
+    'fc' => 'module',
+    'module' => 'mymodulename'
+  ]
+]
+```
+
+### Implementation
+
+This example adds two routes for your module, one for listing items, the other one for showing one specific item: 
```

Comment:
this one is not accurate enough, you've shown how to declare the array, but maybe you should also mention ModuleFrontControllers that developers have to create. It'd be great to provide a full example for this particular hook